sway/desktop/output.c

```diff
@@ -98,27 +98,53 @@ static void output_resolution_notify(struct wl_listener *listener, void *data) {
 	arrange_windows(soutput->swayc, -1, -1);
 }
 
+static void output_scale_notify(struct wl_listener *listener, void *data) {
+	struct sway_output *soutput = wl_container_of(
+			listener, soutput, scale);
+	arrange_windows(soutput->swayc, -1, -1);
+}
+
+static void output_transform_notify(struct wl_listener *listener, void *data) {
```

Comment:
Have you considered not listening to scale/transform/resolution but only output layout changes for determining when to rearrange the windows?